compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilContext.kt

```diff
@@ -0,0 +1,12 @@
+package com.squareup.anvil.compiler.api
+
+/**
+ * Contains context about the given Anvil compiler invocation. Considered a read-only API.
+ *
+ * These loosely correspond to the Anvil Gradle plugin's options, but we may add other information
+ * in the future.
+ */
+public interface AnvilContext {
+  public val generateFactories: Boolean
+  public val generateFactoriesOnly: Boolean
```

Comment:
I think it's worth to copy some parts from `AnvilExtension` to make it clearer what they mean. Sorry, I missed that initially.